Orange/widgets/unsupervised/owpca.py

```diff
@@ -160,6 +174,23 @@ def update_model(self):
         else:
             self.__timer.stop()
 
+    def update_buttons_sparse(self):
```

Comment:
Instead of this I would simply put:
```py
def toggle_buttons(self, sparse_data=False):
    buttons = self.decomposition_box.buttons
    for i, cls in enumerate(DECOMPOSITIONS):
        buttons[i].setDisabled(sparse_data and not cls.supports_sparse)
```

and then call with `self.toggle_buttons(data is not None and data.is_sparse())`